CORE-023 — fix remaining 9 MyPy type errors

## Problem

After fixing 32 MyPy errors in #2473, 9 errors remain in telegram_bot files.

## Remaining errors

```
src/adapters/llm/factory.py:22: error: Item "None" of "str | None" has no attribute "strip"
src/adapters/embeddings/factory.py:24: error: Item "None" of "str | None" has no attribute "strip"
telegram_bot/services/generate_response.py:700: error: Incompatible types in assignment
telegram_bot/services/generate_response.py:704: error: Incompatible types in assignment
telegram_bot/pipelines/client.py:394: error: Argument "state_contract" has incompatible type
telegram_bot/assistant_core_adapter.py:59: error: Returning Any from function
telegram_bot/assistant_core_adapter.py:59: error: "object" not callable
telegram_bot/bot.py:2677: error: Incompatible types in assignment
telegram_bot/bot.py:2715: error: Argument "bot_context" has incompatible type
```

## Files to fix

- `src/adapters/llm/factory.py` — None check
- `src/adapters/embeddings/factory.py` — None check
- `telegram_bot/services/generate_response.py` — type mismatch
- `telegram_bot/pipelines/client.py` — arg type mismatch
- `telegram_bot/assistant_core_adapter.py` — Any return + callable
- `telegram_bot/bot.py` — BotContext/DummyContext mismatch

## Acceptance criteria

- [ ] `uv run mypy src/ telegram_bot/ --ignore-missing-imports` reports 0 errors
- [ ] `make check` passes
- [ ] `make test` passes
- [ ] `make test-contract` passes

## References

- PR #2473 — fixed 32 errors
- Epic: #2411

--- a/src/evaluation/generate_test_queries.py
+++ b/src/evaluation/generate_test_queries.py
@@ -15,7 +15,7 @@ from pydantic import BaseModel, Field
 from qdrant_client import QdrantClient, models
 
 from src.config import Settings
-from src.runtime.llm import create_litellm_chat_client
+from src.runtime.llm import LiteLLMChatClient, create_litellm_chat_client
 
 
 class GeneratedQueries(BaseModel):
@@ -94,7 +94,7 @@ def fetch_article_texts(collection_name: str, article_numbers: list[str]) -> dic
 
 
 async def generate_queries_for_article(
-    client: object, model: str, article_num: str, article_text: str
+    client: LiteLLMChatClient, model: str, article_num: str, article_text: str
 ) -> list[dict]:
     """
     Generate 3 types of queries for a single article.

